pkg/client/fake/client.go

```diff
@@ -1108,10 +1127,28 @@ func (sw *fakeSubResourceClient) Update(ctx context.Context, obj client.Object,
 	updateOptions.ApplyOptions(opts)
 
 	body := obj
-	if updateOptions.SubResourceBody != nil {
-		body = updateOptions.SubResourceBody
+	switch sw.subResource {
+	case "scale":
+		if updateOptions.SubResourceBody == nil {
+			return apierrors.NewBadRequest("expected SubResourceBody")
+		}
+		scale, isScale := updateOptions.SubResourceBody.(*autoscalingv1.Scale)
+		if !isScale {
+			return apierrors.NewBadRequest(fmt.Sprintf("got invalid type %t, expected Scale", updateOptions.SubResourceBody))
```

Comment:
Is this the behavior of the KAS, to first complain with a bad request about the request body before checking if the main resource even exists?

Reply:
I was unable to find it; am I looking at the right place? https://github.com/search?q=repo%3Akubernetes%2Fapiserver+scale+-path%3Atest&type=code

Nonetheless, I was able to reproduce it with `client.Client.Get` and your intuition is correct: it looks up the object first! Great catch.

Reply:
I think the upstream code is here: https://github.com/kubernetes/kubernetes/blob/fb6bbc9781d11a87688c398778525c4e1dcb0f08/pkg/registry/apps/deployment/storage/storage.go#L320

Not exactly a quick read though it seems.